wcwidth/wcwidth.py

```diff
@@ -190,19 +138,36 @@ def wcswidth(pwcs, n=None):
     or -1 if a non-printable character is encountered. When ``n`` is None
     (default), return the length of the entire string.
     """
-    # pylint: disable=C0103
-    #         Invalid argument name "n"
-
-    end = len(pwcs) if n is None else n
-    idx = slice(0, end)
     width = 0
-    for char in pwcs[idx]:
-        wcw = wcwidth(char)
-        if wcw < 0:
-            ucs = ord(char)
-            if _bisearch(ucs, NONZERO_COMBINING):
-                continue
-            return -1
-        else:
-            width += wcw
+    cache = {}
```

Comment:
I understand how cache here can be used to improve a single call to a very large string. Any idea on the memory cost of a well-populated ``global`` cache? Perhaps the wcwidth-browser.py utility could be simply compared resident memory before and after.